File: Problems/13. Roman to Integer.py
"""
Problem Link: https://leetcode.com/problems/roman-to-integer/
"""


class Solution:
    def romanToInt(self, s: str) -> int:
        def romanToIntHelper(str):
            value = 0
            if (str == 'I'):
                value = 1
            elif (str == 'V'):
                value = 5
            elif (str == 'X'):
                value = 10
            elif (str == 'L'):
                value = 50
            elif (str == 'C'):
                value = 100
            elif (str == 'D'):
                value = 500
            elif (str == 'M'):
                value = 1000
            return value

        integer = 0
        for i in range(len(s)):
            if i > 0 and romanToIntHelper(s[i]) > romanToIntHelper(s[i-1]):
                integer += romanToIntHelper(s[i]) - \
                    2 * romanToIntHelper(s[i-1])
            else:
                integer += romanToIntHelper(s[i])

        return integer

File: Problems/13. Roman to Integer.java
/*
Problem Link: https://leetcode.com/problems/roman-to-integer/
*/

class Solution {
    public int romanToIntValue(String s) {
        int value = 0;
        if (s.equalsIgnoreCase("I"))
            value = 1;
        else if (s.equalsIgnoreCase("V"))
            value = 5;
        else if (s.equalsIgnoreCase("X"))
            value = 10;
        else if (s.equalsIgnoreCase("L"))
            value = 50;
        else if (s.equalsIgnoreCase("C"))
            value = 100;
        else if (s.equalsIgnoreCase("D"))
            value = 500;
        else if (s.equalsIgnoreCase("M"))
            value = 1000;
        return value;
    }

    public int romanToInt(String s) {
        int integer = 0;
        for (int i = 0; i < s.length() - 1; i++) {
            int s1 = romanToIntValue(String.valueOf(s.charAt(i)));
            int s2 = romanToIntValue(String.valueOf(s.charAt(i + 1)));
            if (s1 >= s2)
                integer += s1;
            else
                integer -= s1;
        }
        integer += romanToIntValue(String.valueOf(s.charAt(s.length() - 1)));
        return integer;
    }
}